.circleci/config.yml

```diff
@@ -73,21 +95,45 @@ commands:
                 name: Save Yarn Package Cache
                 paths:
                     - ~/.cache/yarn
+    restore_yarn_tools_cache:
+        description: Restores and saves the node_modules directories of the tools package.
+        steps:
+            - restore_cache:
+                key: 2-yarn-{{ arch }}-{{ checksum "tools/yarn.lock" }}-{{ checksum "tools/package.json" }}
+                name: Restore Yarn Tools Package Cache
+            - run:
+                command: yarn --frozen-lockfile --cache-folder ~/.cache/yarn-tools
+                name: Yarn tools
+                working_directory: tools
+            - save_cache:
+                key: 2-yarn-{{ arch }}-{{ checksum "tools/yarn.lock" }}-{{ checksum "tools/package.json" }}
+                name: Save Yarn Tools Package Cache
+                paths:
+                    - ~/.cache/yarn-tools
+                    - tools/node_modules
 jobs:
     build_android:
         docker:
             - image: circleci/android:api-29-node
         environment:
             FASTLANE_SKIP_UPDATE_CHECK: true
-        resource_class: medium+
+        resource_class: medium
         shell: /bin/bash -eo pipefail
         steps:
             - checkout
             - restore_ruby_cache:
                 directory: android
             - restore_yarn_cache
+            - prepare_workspace
+            - restore_environment_variables
+            - run:
+                command: echo "${KEYSTORE_BASE64}" | base64 -di --decode > android/$KEYSTORE_FILENAME.gpg
+                name: Base64 decode keystore env var
             - run:
-                command: bundle exec fastlane build
+                command: sudo gpg --passphrase ${KEYSTORE_ENCRYPTION_KEY} --pinentry-mode loopback -o "android/app/$KEYSTORE_FILENAME" -d "android/$KEYSTORE_FILENAME.gpg"
+                name: Decrypt keystore
+            - run:
```

Comment:
I think we should do this the same as in Integreat. We have a private repository with credentials. There we use openssl to decrypt the keystore. Having the keystore in git has the advantage that we can track it more easily.

We also need this repository for Fastlane Match.

Reply:
Yeah i saw the integreat solution and did a lot of research, which options are possible here and which advantages and disadvantages they have. I like this solutionen more, because CircleCi always keeps the environment varaibles secret and erases them from console, if ever accidentally outputed. But your point of being able to track it is also valid, also i have not thought about Match. For Android I am also thinking about using Google Play App Signing. Have you any experience with it? 

How ever we decide to do it, i would appreciate doing it in a separate ticket / PR or even move it after we done all the iOS App signing stuff.